File: src/e-community/backend/local/e-community-local/e-community-local/Util/Extensions/Python/classes/ConsentContract.py
class ConsentContract:
    """ object for a consent contract (stored on the blockchain) """

    def __init__(self, address_proposer, address_consenter, contract_id, start_e_data, end_e_data,
                 validity_contract, price_hour, price_total, state, data_usage, time_resolution):
        self.address_proposer = address_proposer
        self.address_consenter = address_consenter
        self.contract_id = contract_id
        self.start_e_data = start_e_data
        self.end_e_data = end_e_data
        self.validity_contract = validity_contract
        self.price_hour = price_hour
        self.price_total = price_total
        self.state = state
        self.data_usage = data_usage
        self.time_resolution = time_resolution

    def to_dict(self):
        return {"ContractId": self.contract_id,
                "State": self.state,
                "AddressProposer": self.address_proposer,
                "AddressConsenter": self.address_consenter,
                "StartEnergyData": self.start_e_data,
                "EndEnergyData": self.end_e_data,
                "ValidityOfContract": self.validity_contract,
                "PricePerHour": self.price_hour,
                "TotalPrice": self.price_total,
                "DataUsage": self.data_usage,
                "TimeResolution": self.time_resolution
                }

File: src/e-community/android/compose/app/src/main/java/at/fhooe/ecommunity/data/local/entity/ConsentContract.kt
package at.fhooe.ecommunity.data.local.entity

import androidx.room.Entity
import androidx.room.PrimaryKey
import com.google.gson.annotations.SerializedName
import com.squareup.moshi.Json

/**
 *
 *
 * @param contractId
 * @param state
 * @param addressContract
 * @param addressProposer
 * @param addressConsenter
 * @param startEnergyData
 * @param endEnergyData
 * @param validityOfContract
 * @param pricePerHour
 * @param totalPrice
 */

@Entity(tableName = "contract")
data class ConsentContract (

    @PrimaryKey
    @SerializedName("ContractId")
    var contractId: kotlin.String,

    @SerializedName("State")
    var state: kotlin.Int? = null,

    @SerializedName("AddressContract")
    val addressContract: kotlin.String? = null,

    @SerializedName("AddressProposer")
    val addressProposer: kotlin.String? = null,

    @SerializedName("AddressConsenter")
    val addressConsenter: kotlin.String? = null,

    @SerializedName("StartEnergyData")
    val startEnergyData: kotlin.String? = null,

    @SerializedName("EndEnergyData")
    val endEnergyData: kotlin.String? = null,

    @SerializedName("ValidityOfContract")
    val validityOfContract: kotlin.String? = null,

    @SerializedName("PricePerHour")
    val pricePerHour: kotlin.String? = null,

    @SerializedName("TotalPrice")
    val totalPrice: kotlin.String? = null,

    @SerializedName("DataUsage")
    val dataUsage: kotlin.Int? = null,

    @SerializedName("TimeResolution")
    val timeResolution: kotlin.Int? = null
)